addons/binding/org.openhab.binding.mqtt.generic/src/main/java/org/openhab/binding/mqtt/generic/internal/convention/homeassistant/HaID.java

```diff
@@ -64,62 +68,123 @@ public HaID(String mqttTopic) {
      * @param nodeID The node ID (can be the empty string)
      * @param component The component ID
      */
-    public HaID(String baseTopic, String objectID, String nodeID, String component) {
+    private HaID(String baseTopic, String objectID, String nodeID, String component) {
         this.baseTopic = baseTopic;
         this.objectID = objectID;
         this.nodeID = nodeID;
         this.component = component;
     }
 
-    /**
-     * Creates a {@link HaID} by providing a channel UID.
-     *
-     * @param baseTopic The base topic. Usually "homeassistant".
-     * @param channel The channel UID
-     */
-    public HaID(String baseTopic, ChannelUID channel) {
-        String groupId = channel.getGroupId();
-        if (groupId == null) {
-            throw new IllegalArgumentException("Channel needs a group ID!");
+    public static HaID fromConfig(String baseTopic, Configuration config) {
```

Comment:
The factory pattern is like public constructors -> a fellow developer would really enjoy javadoc here.
There are also new public methods in this class, can you please add javadoc there as well? thanks